### ISSUE TITLE
Suggestion : Super Quick Info fully qualified namespaces for the implemented interfaces

### ISSUE THREAD

**Issue by IneedHelp** [wmjordan/Codist#258]
Would it be possible to add an option which allows displaying fully qualified namespaces for the implemented interfaces as such:

![image](https://github.com/wmjordan/Codist/assets/1366970/f675ddd6-9959-4b3d-a280-b65bcd68b9eb)

**Comment by wmjordan** [comment]
Personally I don't think the namespace before the interface name can be very helpful. What a programmer want to know about an interface is its purpose, thus the name is enough, most of the time. Hence Codist hides namespace names from view as much as possible.
If you do care about the namespace, you can always hover the mouse cursor onto the interface name and a tooltip will tell you that.

**Comment by wmjordan** [comment]
Hmm, this suggestion seems be reasonable.

Current implementation displays interfaces declared/implemented by the type first, then inherited ones.
<img width="433" alt="image" src="https://github.com/wmjordan/Codist/assets/12182730/a98ed87a-f3a8-4c09-b5c7-3e66cebbcd88">
![image](https://github.com/wmjordan/Codist/assets/12182730/f8dc9696-9380-4239-9e7a-8bdd9f05863b)

I don't know whether sorting interfaces by namespaces then by type name can help more.

@fitdev
My friend, can you share your comments here?

**Comment by wmjordan** [comment]
Here's how it looks like when interfaces are sorted by namespace first.

**Comment by IneedHelp** [comment]
I particularly very much like the sorted by namespace first and fully qualified presentation (and in conjunction with no width limit on Super Quick Info as I dislike having word wrap, I feel like it messes with my ability to quickly perceive the information).

Since this is a more visual tweak than other features, it might be a good idea to have it as a toggle option (as an opt-in by default) so that other users don't have to change settings. Or even better, maybe a toggle implemented straight into the Super Quick Info UI; it could be placed on this line:

![image](https://github.com/wmjordan/Codist/assets/1366970/737af83f-8c14-478e-a34c-50db8d4ebd20)

so that nos pace is wasted, since that line always says "Interface:" anyway, even when there's only one interface implemented.

**Comment by wmjordan** [comment]
> I still believe that having a toggle button integrated straight into the UI would be better

Years ago I was trying to put some toggle buttons on the Quick Info to allow changing its content after display, but unfortunately the Quick Info disappeared after it was changed.
I will try again to see whether I can find a way.

**Comment by IneedHelp** [comment]
> I will try again to see whether I can find a way.

Even if toggling closes Quick Info, the state should be remembered in subsequent displays of Quick Info, so in a way it's a faster path to change appearance without having to access the Options window.

**Comment by fitdev** [comment]
> My friend, can you share your comments here?

Personally, fully qualifying interfaces is not something I would find useful, but I understand that for some developers it may be useful.

In any case, I hope that it can be an option which users can customize - turn on or off.

As a middle ground, perhaps interfaces can be _automatically fully qualified_ when there is _more than one_ identically named interface coming from different namespaces:

```
class C : NS1.IFoo, NS2.IFoo {  }
```

In other words fully qualify to avoid ambiguity. Alternatively, perhaps only the immediate parent namespace may be displayed.

**Comment by wmjordan** [comment]
@fitdev 
Thank you for your opinion.

> interfaces can be automatically fully qualified when there is more than one identically named interface coming from different namespaces
> perhaps only the immediate parent namespace may be displayed.

I agree.

**Comment by IneedHelp** [comment]
As some extra reasoning why I suggested this is also because some developers (myself included) employ a top-to-bottom architecture, where dependency increases from top to bottom, and that combined with a proper naming scheme can expose architectural dependencies at a glance, and namespaces play a big role in this organizational strategy. Therefore the grouping or alphabetical display of fully qualified interfaces would present a level of usefulness.

**Comment by wmjordan** [comment]
Although the code is already ready, what is left is only an option.
I still don't like the fully qualified interface names.

The namespaces add little value to my programming experience. I don't care its namespace at all. I can not even remember any second I have ever wanted to know about it before. @fitdev's example maybe the exception. 

But I do care for the functionality of the interface. On the above screenshot of the generic Dictionary, the namespaces take too much space and interfere reading the interface names. It is adding reversal value to programming experience.

**Comment by IneedHelp** [comment]
I could go on further trying to make a case in favor of fully qualified interface names and without backup from other developers using similar architecture strategies as me, it might seem like I just want this for myself. Finally, it's your decision whether you want to add this implementation or not. Although if you do decide to add it as an option, I'd appreciate it. And I believe that keeping both options won't become a maintenance struggle as they aren't fundamentally different.


Also, I understand that you don't value namespaces in your workflow, but for me (and I assume other programmers too) they are very useful in keeping a well structured solution (especially when using folders to organize project files, and those folders' names are reflected in the namespaces).

Here's a small sample of what I mean

![image](https://github.com/wmjordan/Codist/assets/1366970/ee73d3dc-4e08-4794-bde0-a988620d9293)

where basically usages are grouped by project and displayed alphabetically, immediately allowing to discover dependency breaking rules, how heavy a type depends on other types. And all this would similarly translate to how useful it would be viewing implemented interfaces preceded by their full namespaces.

**Comment by wmjordan** [comment]
I am still not convinced.

> for me (and I assume other programmers too) they are very useful in keeping a well structured solution (especially when using folders to organize project files, and those folders' names are reflected in the namespaces).

Why do you care for folders or folder names when you see interface names?
When the code is compiled and used elsewhere, there's no more "folders". What you should care for an interface is the implementation contract, defined by that interface.
When you use `IEnumerable<T>`, do you have a single second care that it comes from `System.Collections.Generics`, but not `System.Collections` or `System.Collections.ReadOnly`?

> basically usages are grouped by project and displayed alphabetically, immediately allowing to discover dependency breaking rules, how heavy a type depends on other types.

I can not think of a typical usage pattern for the above statement.

**Comment by IneedHelp** [comment]
Some of the features I suggest on Codist is stuff I see and use frequently from other tools. Whether I am the only one that finds it useful or a lot many that haven't yet discovered Codist, I don't know. Further attempts to create a compelling argument are pointless in this case, and I understand you have your own vision for your tool, and I respect that.

**Comment by wmjordan** [comment]
The code is in [the previous commit](https://github.com/wmjordan/Codist/commit/4118a22784a752556f594565576ca7ac521430c8).
